src/Http/Http/src/Timeouts/RequestTimeoutAttribute.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+/// <summary>
+/// Metadata that provides endpoint-specific request timeouts.
+/// </summary>
+/// <remarks>
+/// The default policy will be ignored with this attribute applied.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+public sealed class RequestTimeoutAttribute : Attribute
+{
+    /// <summary>
+    /// The timeout to apply for this endpoint.
+    /// </summary>
+    public TimeSpan? Timeout { get; }
+
+    /// <summary>
+    /// The name of the policy which needs to be applied.
+    /// This value is case insensitve.
+    /// </summary>
+    public string? PolicyName { get; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="RequestTimeoutAttribute"/> using the specified timeout.
+    /// </summary>
+    /// <param name="milliseconds">The amount of timeout for this specific endpoint.</param>
+    public RequestTimeoutAttribute(int milliseconds)
```

Comment:
@halter73, @JamesNK, this constructor has been bugging me because it's got awful usability. What about this instead:
```suggestion
    public RequestTimeoutAttribute(int milliseconds = 0, int seconds = 0, int minutes = 0)
```
I know the larger unit should go first, but I don't want RequestTimeoutAttribute(1) to default to minutes. Ideally it would be seconds, but putting them out of order seems even weirder.

Alternatively, we say millisecond timeouts are uncommon enough that you should specify them via policy.
```suggestion
    public RequestTimeoutAttribute(int seconds, int minutes = 0)
```

We'd also need to throw an ArgumentException if they were all unspecified (0) or the total was negative.

Thoughts?

Reply:
Discussion notes: It's fine, we don't want to complicate the API. In the future we could consider a no-param constructor and properties for Seconds, Minutes, etc.